test/amplitude-client.js

```diff
@@ -285,6 +285,20 @@ describe('AmplitudeClient', function () {
       amplitude._getUrlParams.restore();
     });
 
+    it('should reject invalid device ids that contain periods', function () {
+      const spyErrorWarning = sinon.spy(utils.log, 'error');
+      const badDeviceId = 'bad.device.id';
+      amplitude.init(apiKey, null, { deviceId: badDeviceId });
+      assert.isTrue(
+        spyErrorWarning.calledWith(
+          `Invalid device ID rejected. Randomly generated UUID will be used instead of "${badDeviceId}"`,
+        ),
+      );
+      assert.notEqual(amplitude.options.deviceId, badDeviceId);
```

Comment:
in this case, users will also expect to get the warning that they specifically can't use a period in the deviceId, right?

Reply:
correct, I also added another assert for this